DOCUMENTATION.md

```diff
@@ -2180,11 +2188,266 @@ const value = await map.get('key');
+await map.put('key2', 2);
+await map.put('key3', 3);
+
+const result = client.getSqlService().execute(`SELECT __key, this FROM my-distributed-map WHERE this > 1`);
+
+for await(const row of result){
+    console.log(row); // {__key: 'key3', this: 3} and {__key: 'key2', this: 2}
+}
+```
+
+### Casting
+
+When comparing a column with a parameter, your parameter must be of a compatible type. Since Node.js client uses double
+as default number type, to compare with an integer based column you can use long objects or casting.
+
+The similar thing applies to other data types. You can cast string to every SQL type.
+
+#### Using long
+
+In the example below, age column is of type `INTEGER`. Since long objects are sent as `BIGINT`, there is no problem in the
+comparison.
+
+```javascript
+const result = client.getSqlService().execute(
+    'SELECT * FROM myMap WHERE age > ? AND age < ?',
+    [long.fromNumber(13), long.fromNumber(18)]
+);
+```
+
+#### Using casting
+
+In the example below, age column is of type `INTEGER`. The default number type is `double` by default. Since we cast
+doubles as `BIGINT`, there is no problem in the comparison. Note that we can also cast to other types that are be
+comparable with `INTEGER`.
+
+```javascript
+const result = client.getSqlService().execute(
+    'SELECT * FROM myMap WHERE age > CAST(? AS BIGINT) AND age < CAST(? AS BIGINT)',
+    [13, 18]
+);
+```
+
+### 8.7.1. Querying IMap
+
+The following subsections describe how you can access Hazelcast map objects and perform queries on them.
+
+#### Names
+
+The SQL service exposes `IMap` objects as tables in the predefined `partitioned` schema using exact names.
+This schema is in the SQL service search path so that you can access the `IMap` objects with or without the schema name.
+
+Schema and table names are case-sensitive; you can access the `employee` map, for example, as employee or `partitioned.employee`,
+but not as `Employee`:
+```sql
+SELECT * FROM employee
+SELECT * FROM partitioned.employee
+```
+
+#### Field
+
+The SQL service resolves fields accessible from the SQL automatically. The service reads the first local entry pair of
+the `IMap` to construct the list of fields. If the `IMap` does not have local entries on the member where the query is
+started, then the list of fields cannot be resolved, and an exception is thrown.
+
+Field names are case-sensitive.
+
+
+##### Key and Value Objects
+
+An `IMap` entry consists of a key and a value. These are accessible through the `__key` and `this` aliases. The following
+query returns the keys and values of all entries in a map:
+
+```sql
+SELECT __key, this FROM employee
+```
+
+##### Key and Value Fields
+
+You may also access the nested fields of a key or a value. The list of exposed fields depends on the serialization format, as described below:
+* For [IdentifiedDataSerializable](#41-identifieddataserializable-serialization) objects, you can use public field name or getter names.
```

Comment:
I think there should be a new line before this bulleted list item (to render it properly).